move role off of label and onto child span el

erikphansen opened the pull request:
## Description
move `role` and interaction-related attributes off of `label` tag and onto a new child `span` element.

**Note that this change changes the appearance of the focus outline as the outline surrounds the child span now rather than the parent label. See screenshot.**

## Testing done
- confirmed locally that the file upload component continues to work as intended

## Screenshots
![image](https://user-images.githubusercontent.com/20728956/60826623-19e84f00-a163-11e9-80de-48981fbef5de.png)

## Acceptance criteria
- [ ] aXe scanner — and Trevor — are happy
- [x] file uploading still works as intended

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

erikphansen committed:
move role off of label and onto child span el

erikphansen committed:
fix test

jbalboni commented:
@erikphansen Does this work in IE11?

erikphansen commented:
I... hope so. Will check...

cvalarida commented:
I just tested in IE11 and it looks great. Keyboard interaction works too. Tabbing, spacebar, return, the works. :+1:

The only kinda wonky thing is it highlights the words inside the button, not the whole button (as shown in the screenshot above).

erikphansen commented:
FWIW, I think we could keep the highlighting on the button (err, `label`) itself. But I got an a11y ESLint error when I removed the `role="button"` but left the `tabIndex` and `onKeyPress`, which is why I moved those to the new `span` as well.
